### PR TITLE
docs(bpmn): explain binding types for call activities, business rule tasks, and user tasks

### PR DESCRIPTION
**Pull request by jfriedenstab**
## Description
Add documentation on the different binding types for linked resources: `deployment` (new type introduced with `8.6.0-alpha4`) vs. `latest`.

Closes https://github.com/camunda/camunda-docs/issues/4124
Part of https://github.com/camunda/product-hub/issues/1920

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [x] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).
- [x] I included my new page in the sidebar file(s).
- [ ] I added a redirect for a renamed or deleted page to the .htaccess file.

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

File: docs/components/best-practices/modeling/choosing-the-resource-binding-type.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+- [call activities](/docs/components/modeler/bpmn/call-activities/call-activities.md#defining-the-called-process)
+- [business rule tasks](/docs/components/modeler/bpmn/business-rule-tasks/business-rule-tasks.md#defining-a-called-decision) (if the DMN decision implementation is used)
+- [user tasks](/docs/components/modeler/bpmn/user-tasks/user-tasks.md#user-task-forms) (if a Camunda Form is linked)
+
+The binding type determines the version of the target resource that will be used at runtime.
+For example, in case of a call activity, it would be the version of the called process to be instantiated.
+
+The following binding types are currently supported in Camunda 8:
+
+- **`latest`** will resolve to the **latest deployed version** of the target resource (at the point in time the process element is activated).
```
Comment (HanselIdes) [inline docs/components/best-practices/modeling/choosing-the-resource-binding-type.md:19]:
```suggestion
- **`latest`** will resolve to the **latest deployed version** of the target resource at the moment the process element is activated.
```

File: docs/components/best-practices/modeling/choosing-the-resource-binding-type.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+---
+title: Choosing the resource binding type
+tags:
+  - BPMN
+description: "Understand the differences between 'latest' and 'deployment' binding for linked resources"
+---
+
```
Comment (HanselIdes) [inline docs/components/best-practices/modeling/choosing-the-resource-binding-type.md:7]: I like starting pages with a value proposition to answer "why should I care?"
```suggestion
Camunda offers version binding for linked processes, decisions, or forms so you can deploy new versions without disrupting live processes and prevent production outages.
```

File: docs/components/best-practices/modeling/choosing-the-resource-binding-type.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+The binding type determines the version of the target resource that will be used at runtime.
+For example, in case of a call activity, it would be the version of the called process to be instantiated.
+
+The following binding types are currently supported in Camunda 8:
+
+- **`latest`** will resolve to the **latest deployed version** of the target resource (at the point in time the process element is activated).
+  Be aware that this can lead to unexpected behavior if you deploy a new version of the target resource without ensuring backwards compatibility with every deployed process that depends on it.
+  Therefore, using `latest` might not be suited for production environments that require stability and a predictable behavior.
+
+- **`deployment`** will resolve to the particular version of the target resource that was **deployed together** with the currently running version of the process in the **same deployment**.
```
Comment (HanselIdes) [inline docs/components/best-practices/modeling/choosing-the-resource-binding-type.md:23]: A user might wonder: "how do I deploy multiple resources in the same deployment"
Answer: Use a process application's deployment UI or use the Zeebe deployment API with multiple files (needs to be revised for technical accuracy)
 
Links to these docs pages would be helpful
https://docs.camunda.io/docs/apis-tools/zeebe-api/gateway-service/#deployresource-rpc
https://docs.camunda.io/docs/next/components/modeler/web-modeler/process-applications/

**Review by HanselIdes — CHANGES_REQUESTED** [review]
Hi @jfriedenstab , I have a few edits for the user journey and value proposition but overall I'm happy with it.
Besides this, I ignored some of the writing style changes that a technical writer would pick up

**Comment by mesellings** [comment]
@jfriedenstab I will take a look at this for you 👍

File: docs/components/best-practices/modeling/choosing-the-resource-binding-type.md (hunk 3 above)
Comment (jfriedenstab) [reply docs/components/best-practices/modeling/choosing-the-resource-binding-type.md:23]: Thanks, good suggestion! 👍🏻

**Comment by jfriedenstab** [comment]
Thanks for the review @HanselIdes! I incorporated your suggestions.

File: docs/components/best-practices/modeling/choosing-the-resource-binding-type.md (hunk 2 above)
Comment (jfriedenstab) [reply docs/components/best-practices/modeling/choosing-the-resource-binding-type.md:7]: Ah, I just noticed that you meant this to be the first sentence of the page, not the `description` tag.

**Review by HanselIdes — DISMISSED** [review]
LGTM

**Comment by mesellings** [comment]
I will take a look now for you @jfriedenstab

**Review by mesellings — DISMISSED** [review]
Hi @jfriedenstab, I have performed a TW review and commit, focusing mainly on grammar, styles, etc. I have made some edits for these, such as using active voice and so on, and am happy to approve this work from a writing standpoint, although I cannot vouch for the technical details obviously.

For the main page I think the supported binding types should be in a table, but as it is not possible currently in our Docusaurus implementation to mix markdown tables with lists and versioned hyperlinks, I have left them as lists for now.

**Review by mesellings — APPROVED** [review]
lgtm

**Review by korthout — COMMENTED** [review]
@jfriedenstab Sorry for late review. 

👍 I really like the best practices guide
👍 I like how you link to the guide from the bpmn docs
👍 I like how you've described the binding type in the bpmn docs

❌ I'd like to an addition to 'choosing the resource binding type' so users can make a meaningful decision between the two types.

🔧 I also have a minor suggestion regarding the XML examples, but  feel free to ignore if you disagree.

File: docs/components/best-practices/modeling/choosing-the-resource-binding-type.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+| `latest`     | <p>Resolves to the **latest deployed version** of the target resource at the moment the process element is activated.</p><p><ul><li><p>This can lead to unexpected behavior if you deploy a new version of the target resource without ensuring backwards compatibility with every deployed process that depends on it.</p></li><li><p>Therefore, using `latest` might not be suited for production environments that require stability and a predictable behavior.</p></li></ul></p>                                                                                                                                                                                                                                                                                                                                                                           |
```
[the remark this answers is not in the document]
Comment (jfriedenstab) [reply docs/components/best-practices/modeling/choosing-the-resource-binding-type.md:26]: Thanks, @korthout, for your review. I've finally opened a follow-up PR to address your comments: https://github.com/camunda/camunda-docs/pull/4177

**Pull request merged**